Docs build fail due to request timeouts

**Describe the bug**
It happens from time to time that the builds fail because the link checker got an timeout. This is kind of annoying, because we have to run the complete test suite again. Maybe we can do it in a earlier state and also increase the timeout. Or simple force to fail the pipeline. Currently we have to wait until the test suite is done and even if the test are green then the pipeline will fail because of this timeout.

```

+ .ci/scripts/docs/build.sh

+ export MDBOOK_OUTPUT__LINKCHECK__FOLLOW_WEB_LINKS=true

+ cd docs/

+ mdbook build

2019-11-21 08:01:02 [INFO] (mdbook::book): Book building has started

Replacing commit template {{commit}} with hash aa2ea4c5c83015899d66901ab2fc2984eb273736

2019-11-21 08:01:03 [INFO] (mdbook::book): Running the html backend

Replacing commit template {{commit}} with hash aa2ea4c5c83015899d66901ab2fc2984eb273736

2019-11-21 08:01:08 [INFO] (mdbook::book): Running the linkcheck backend

2019-11-21 08:01:08 [INFO] (mdbook::renderer): Invoking the "linkcheck" renderer

error: Unable to retrieve "https://maven.apache.org/": https://maven.apache.org/: timed out



    ┌── java-client/get-started.md:21:3 ───

    │

 21 │ * [Apache Maven](https://maven.apache.org/)

    │   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ https://maven.apache.org/: timed out

    │



Error: ErrorMessage { msg: "One or more incorrect links" }

2019-11-21 08:01:41 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.

2019-11-21 08:01:41 [ERROR] (mdbook::utils): Error: Rendering failed

2019-11-21 08:01:41 [ERROR] (mdbook::utils): 	Caused By: The "linkcheck" renderer failed

script returned exit code 101
```

or 

```

+ .ci/scripts/docs/build.sh

+ export MDBOOK_OUTPUT__LINKCHECK__FOLLOW_WEB_LINKS=true

+ cd docs/

+ mdbook build

2019-11-21 07:09:08 [INFO] (mdbook::book): Book building has started

Replacing commit template {{commit}} with hash 777c0da3827690eb847d5bccdf0d69294adfa142

2019-11-21 07:09:09 [INFO] (mdbook::book): Running the html backend

Replacing commit template {{commit}} with hash 777c0da3827690eb847d5bccdf0d69294adfa142

2019-11-21 07:09:14 [INFO] (mdbook::book): Running the linkcheck backend

2019-11-21 07:09:14 [INFO] (mdbook::renderer): Invoking the "linkcheck" renderer

error: Unable to retrieve "http://www.bpmn.org/": http://www.bpmn.org/: error trying to connect: failed to lookup address information: Name or service not known



    ┌── basics/workflows.md:51:12 ───

    │

 51 │ Zeebe uses [BPMN 2.0](http://www.bpmn.org/) for representing workflows. BPMN is an industry standard which is widely supported by different vendors and implementations. Using BPMN ensures that workflows can be interchanged between Zeebe and other workflow systems.

    │            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ http://www.bpmn.org/: error trying to connect: failed to lookup address information: Name or service not known

    │



Error: ErrorMessage { msg: "One or more incorrect links" }

2019-11-21 07:09:19 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.

2019-11-21 07:09:19 [ERROR] (mdbook::utils): Error: Rendering failed

2019-11-21 07:09:19 [ERROR] (mdbook::utils): 	Caused By: The "linkcheck" renderer failed

script returned exit code 101
```

--- a/.ci/scripts/docs/build.sh
+++ b/.ci/scripts/docs/build.sh
@@ -1,7 +1,7 @@
 #!/bin/sh -eux
 
 # enable external link checker
-#export MDBOOK_OUTPUT__LINKCHECK__FOLLOW_WEB_LINKS='true'
+export MDBOOK_OUTPUT__LINKCHECK__FOLLOW_WEB_LINKS='true'
 
 cd docs/
 mdbook build

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -176,10 +176,12 @@ pipeline {
 
                 stage('Build Docs') {
                     steps {
+                      retry(3) {
                         container('maven') {
                             sh '.ci/scripts/docs/prepare.sh'
                             sh '.ci/scripts/docs/build.sh'
                         }
+                      }
                     }
                 }
             }

